src/core/ext/xds/xds_common_types.cc

```diff
@@ -45,6 +46,25 @@
 
 namespace grpc_core {
 
+//
+// ParseDuration()
+//
+
+Duration ParseDuration(const google_protobuf_Duration* proto_duration,
+                       ValidationErrors* errors) {
+  int64_t seconds = google_protobuf_Duration_seconds(proto_duration);
+  if (seconds < 0 || seconds > 315576000000) {
```

Comment:
we are using these constants at multiple places. Let's define them in one place and use that name everywhere

Reply:
We're using them in exactly two places, here and in the JSON parsing code.  I don't really see a convenient common dependency where we could put the constant, and I don't think the duplication is going to cause enough of a problem to warrant the overhead of creating a new module just for this constant.